pkgs/applications/networking/cluster/helm/plugins/helm-dt.nix

```diff
@@ -0,0 +1,47 @@
+{
+  buildGoModule,
+  fetchFromGitHub,
+  lib,
+  ...
+}:
+buildGoModule rec {
+  pname = "helm-dt";
+  version = "0.4.1";
+
+  src = fetchFromGitHub {
+    owner = "vmware-labs";
+    repo = "distribution-tooling-for-helm";
+    rev = "v${version}";
+    sha256 = "sha256-KrQAlB0ORNzKIG2vxych3gVBytTh3Hhnjsyn1ia1ZQM=";
+  };
+
+  vendorHash = "sha256-T8Kk+9NAhYOvSq94HOEE53BT7Xh9tU1gJ420o/tiVEo=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X 'main.BuildDate=1970-01-01 00:00:00 UTC'"
+    "-X 'main.Commit=v${version}'"
+  ];
+
+  # NOTE: Remove the install and upgrade hooks.
+  postPatch = ''
+    sed -i '/^hooks:/,+2 d' plugin.yaml
+  '';
+
+  doCheck = false;
+  CGO_ENABLED = 1;
+
+  postInstall = ''
+    install -dm755 $out/helm-dt/bin
+    mv $out/bin/dt $out/helm-dt/bin/dt
+    rmdir $out/bin
```

Comment:
Wait are you sure to remove `$out/bin`?

Reply:
The path that plugin expects is under `helm-dt/bin/` instead of `bin/`

Reply:
nixpkgs convention are quite strict when it comes to derivation content, and binaries are expected to be located within `$out/bin`. Is it possible for you to patch this?

Reply:
I will check, thank you

Reply:
So `helm` requires that "plug-ins" have a folder structure with the plug-in name, `helm-dt`, a file named `plugins.yaml` inside. Per official [docs](https://helm.sh/docs/topics/plugins/#building-plugins)

The plug-ins themselves really require the use of the `wrapHelm` function, which like the below example:

```nix
{
  environment.systemPackages = with pkgs; [
    (wrapHelm kubernetes-helm {
      plugins = with pkgs.kubernetes-helmPlugins; [
        helm-secrets
        helm-diff
        helm-s3
        helm-git
      ];
    }) 
  ];
}
```

Creates a package with the following contents:

```bash
$ ls /nix/store/*-helm-3.15.3
bin  helm-diff  share

$ ls /nix/store/*-helm-3.15.3/helm-diff
bin  plugin.yaml
```

So it is not really possible for me to simply refactor it to be more inline with the nix style

Existing:
- [`helm-unittest`](https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/applications/networking/cluster/helm/plugins/helm-unittest.nix#L22-L25)
- [`helm-s3`](https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/applications/networking/cluster/helm/plugins/helm-s3.nix#L31-L33)

Reply:
Well that's a bit odd, but i guess it's fine then
So many exceptions are living in nixpkgs :sweat_smile:

Reply:
Indeed, thank you for the code suggestions, working on them well I was researching that